include/nbl/video/IDeviceMemoryBacked.h

```diff
@@ -1,44 +1,75 @@
-// Copyright (C) 2018-2020 - DevSH Graphics Programming Sp. z O.O.
+// Copyright (C) 2018-2022 - DevSH Graphics Programming Sp. z O.O.
 // This file is part of the "Nabla Engine".
 // For conditions of distribution and use, see copyright notice in nabla.h
-
-#ifndef __NBL_I_DRIVER_MEMORY_BACKED_H_INCLUDED__
-#define __NBL_I_DRIVER_MEMORY_BACKED_H_INCLUDED__
+#ifndef _NBL_I_DRIVER_MEMORY_BACKED_H_INCLUDED_
+#define _NBL_I_DRIVER_MEMORY_BACKED_H_INCLUDED_
 
 #include "IDeviceMemoryAllocation.h"
 #include <algorithm>
 
-#define VK_NO_PROTOTYPES
-#include <vulkan/vulkan.h>
-
 namespace nbl::video
 {
 
-//! Interface from which resources backed by IDeviceMemoryAllocation, such as ITexture and IGPUBuffer, inherit from
+//! Interface from which resources backed by IDeviceMemoryAllocation inherit from
 class IDeviceMemoryBacked : public virtual core::IReferenceCounted
 {
     public:
+        //! If you bound an "exotic" memory object to the resource, you might require "special" cleanups in the destructor
+        struct ICleanup
+        {
+            virtual ~ICleanup() = 0;
+        };
+
+        //!
+		struct SCachedCreationParams
+		{
+            // A Pre-Destroy-Step is called out just before a `vkDestory` or `glDelete`, this is only useful for "imported" resources
+            std::unique_ptr<ICleanup> preDestroyStep = nullptr;
```

Comment:
I will rename this to `preDestroyCleanup`,  same for the other

Reply:
fine by me